### ISSUE TITLE
False positive with CA1508 and IEnumerator<T>.Current

### ISSUE THREAD

**Issue by Crono1981** [dotnet/roslyn-analyzers#4936]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508): `Avoid dead conditional code`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.100](https://dotnet.microsoft.com/download/dotnet/5.0)

### Describe the bug

Analyzer reports a false positive if checking for a null value in `IEnumerator<T>.Current` property.

### Steps To Reproduce

Have the analyzer go through the following C# code:

```cs
using System;
using System.Collections.Generic;

public class Program
{
	public static void Main()
	{
		IEnumerable<string> list = new[] {"val0", null, "val2"};
		
		using var enumer = list.GetEnumerator();
		
		while (enumer.MoveNext())
		{
			var val = enumer.Current;
			
			if (val != null)
			{
				Console.WriteLine(val);
			}
		}
	}
}
```

### Expected behavior

Analyzer should acknowledge that there's indeed a possibility that `val` on line 16 may be `null` and not raise any issues about it.

### Actual behavior

Analyzer wrongfully reports a CA1508 issue on line 16, saying that `val != null` is always `true`.

**Comment by mavasani** [comment]
Need to verify on 5.0.3 NetAnalyzers NuGet package

**Comment by mavasani** [comment]
This does not repro on the latest .NET 8 preview SDK
